nni/compression/pytorch/utils/counter.py

```diff
@@ -1,18 +1,217 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
+import functools
+from collections import defaultdict
+from prettytable import PrettyTable
 import torch
 import torch.nn as nn
 from nni.compression.pytorch.compressor import PrunerModuleWrapper
 
-try:
-    from thop import profile
-except Exception as e:
-    print('thop is not found, please install the python package: thop')
-    raise
 
+def _get_params(m):
+    return sum([p.numel() for p in m.parameters()])
 
-def count_flops_params(model: nn.Module, input_size, custom_ops=None, verbose=True):
+def count_convNd(m, x, y, add_results):
+    cin = m.in_channels
+    kernel_ops = m.weight.size()[2] * m.weight.size()[3]
+    output_size = torch.zeros(y.size()[2:]).numel()
+    cout = y.size()[1]
+
+    if hasattr(m, "weight_mask"):
+        cout = m.weight_mask.sum() // (cin * kernel_ops)
+
+    total_ops = cout * output_size * kernel_ops * cin // m.groups  # cout x oW x oH
+
+    results = {
+        "flops": total_ops,
+        "params": _get_params(m),
+        "weight_size": tuple(m.weight.size()),
+        "input_size": tuple(x[0].size()),
+        "output_size": tuple(y.size()),
+        "module_type": type(m).__name__
+    }
+
+    add_results(m._name, **results)
```

Comment:
How about `count_xxx` functions just return the results, and do the `add_results` outside? it will make `count_xxx` functions cleaner.

Reply:
Thanks. I have updated the code, please review the latest version~

Reply:
`countxxx` still calls `self._push_results`,   I mean define another common hook function for example `myhook` to call `countxxx` by module type, then do `_push_results` in `myhook`.  Register my hook as the hook using something like `m.register_forward_hook(myhook)`,  in this way `countxxx` is a pure util function with single responsibility and can be reused later.

Reply:
fix it~